### ISSUE TITLE
Missing warning or documentation for upcast in sparse_softmax_cross_entropy_with_logits?

### ISSUE THREAD

**Issue by anshulsamar** [tensorflow/tensorflow#26417]
In the "sparse_softmax_cross_entropy_with_logits" function inside tensorflow/python/ops/nn_ops.py, logits that are in fp16 are automatically upcasted to fp32. At the end of the function, the result is cast back down to fp16. 

This functionality ought to be exposed in documentation and/or a warning. For users doing mixed precision training, the expected behavior would be that the entire loss (softmax + cross entropy) is done in fp16. 

Relevant lines:
```
logits = ops.convert_to_tensor(logits)
precise_logits = math_ops.cast(logits, dtypes.float32) if (dtypes.as_dtype(
	logits.dtype) == dtypes.float16) else logits
...
cost, _ = gen_nn_ops.sparse_softmax_cross_entropy_with_logits(
	precise_logits, labels, name=name)
```

Thanks so much.

**Comment by jvishnuvardhan** [comment]
@anshulsamar I know this is a stale issue. There were lot of improvements with `mixed_precision`.  Is this still an issue for you? 

If this is still an issue, can you please share a simple standalone code to reproduce the issue. Thanks!
